satpy/readers/netcdf_utils.py

```diff
@@ -380,6 +374,23 @@ def _compose_replacement_names(variable_name_replacements, var, variable_names):
                 variable_names.append(var.format(**{key: val}))
 
 
+def get_data_as_xarray(variable):
+    """Get data in variable as xr.DataArray."""
+    try:
+        attrs = variable.attrs
+    except AttributeError:
+        attrs = variable.__dict__
+    try:
+        arr = xr.DataArray(
+            variable[:], dims=variable.dimensions, attrs=attrs, name=variable.name)
+    except ValueError:
```

Comment:
Just curious, does the `ValueError` come from the `variable[:]`? Would it make more sense to have a `try/except` dedicated to just the array creation and then a single `xr.DataArray(...)` creation at the end (that uses the `attrs` and `var_arr` or whatever created from the previous try/excepts).

Reply:
Yes, that's it. I refactored as suggested and added a test for the scalar case in https://github.com/pytroll/satpy/pull/2753/commits/12854de2dc6bb09cb8a90d49d5e15e1aeb92a411